pkgs/os-specific/linux/mbp-modules/macbook15-spi-driver/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, kernel, fetchFromGitHub, }:
+
+stdenv.mkDerivation rec {
+  name = "macbook15-spi-driver";
+  version = "0.1";
+
+  src = fetchFromGitHub {
+    owner = "hlolli";
+    repo = "macbook15-spi-driver";
+    rev = "86a2f37ca0d2a4729a32b636dc48f39c522dd8ad";
+    sha256 = "1724s3y8ad14s8a0rag9jadb69j0j2vn566fzg6basx2bmfrj8gi";
+  };
+
+  buildPhase = ''
+    make -C ${kernel.dev}/lib/modules/${kernel.modDirVersion}/build \
```

Comment:
You can do `makeFlags = [ "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build" ];` here: https://github.com/hlolli/macbook15-spi-driver/blob/86a2f37ca0d2a4729a32b636dc48f39c522dd8ad/Makefile#L9

also see `nixpkgs/pkgs/os-specific/linux/exfat/default.nix`